File: Server/core/src/main/java/com/ttrip/core/repository/landmark/LandmarkRepository.java
package com.ttrip.core.repository.landmark;

import com.ttrip.core.entity.landmark.Landmark;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface LandmarkRepository extends JpaRepository<Landmark, Integer> {
    Optional<Landmark> findByLandmarkId(Integer id);
}

File: Ttrip/app/src/main/java/org/sfy/ttrip/domain/repository/landmark/LandmarkRepository.kt
package org.sfy.ttrip.domain.repository.landmark

import org.sfy.ttrip.data.remote.Resource
import org.sfy.ttrip.domain.entity.landmark.BadgeItem
import org.sfy.ttrip.domain.entity.landmark.LandmarkItem

interface LandmarkRepository {

    suspend fun getLandmarks(): Resource<List<LandmarkItem>>

    suspend fun getBadges(): Resource<List<BadgeItem>>

    suspend fun issueBadges(landmarkId: Int): Int
}